web/src/widget_modal.ts

```diff
@@ -68,3 +80,39 @@ export function frame_poll_message_content(): string {
         .filter(Boolean);
     return "/poll " + question + "\n" + options.join("\n");
 }
+
+export function todo_list_tasks_setup(): void {
+    const $todo_options_list = $("#add-todo-form .todo-options-list");
+    $todo_options_list.on("input", "input.todo-input", (event) => {
+        add_option_row(event, "TODO");
+    });
+    $todo_options_list.on("input", "input.todo-description-input", (event) => {
+        add_option_row(event, "TODO");
+    });
+    $todo_options_list.on("click", "button.delete-option", delete_option_row);
+
+    setup_sortable_list("#add-todo-form .todo-options-list");
+}
+
+export function frame_todo_message_content(): string {
+    const title = $<HTMLInputElement>("input#todo-title-input").val()?.toString().trim() ?? "";
+    const todo_str = title ? `/todo ${title}\n` : "/todo Task list\n";
+
+    const todos: string[] = [];
+
+    $(".option-row").each(function () {
+        const todo_name = $(this).find(".todo-input").val()?.toString().trim() ?? "";
+        const todo_description =
+            $(this).find(".todo-description-input").val()?.toString().trim() ?? "";
+
+        if (todo_name || todo_description) {
+            const todo =
+                todo_name && todo_description
+                    ? `${todo_name}: ${todo_description}`
+                    : todo_name || todo_description;
```

Comment:
Again, this is really hard to read. If I understand correctly, in the absence of a task title, we assign it the value in the description. But that's not the behavior we want. 

We should aim for consistency between this modal UI and the pre-existing widget UI. Since task titles are mandatory, we do not want to consider tasks without task title. To discourage only filling in the description field, we should considering enabling the input only when the corresponding task title input is non empty. In case you like, feel free to discuss this detail more on CZO

Reply:
currently we have two option:

1. Enable the description input only when the task title has values:
![image](https://github.com/zulip/zulip/assets/73663475/f8ad3c07-adbe-4102-98ce-252c41d377d7)
2. Stay consistent with the current widget (which is displayed after the /todo command is sent) in which you can add a value to the description, but the "Add task" button will not work.

https://github.com/zulip/zulip/assets/73663475/95461a76-fe70-49f0-aff1-2629a2c1dc1e



- In my opinion, what you have suggested sounds better than the second option. We should not allow users to write values in the description if the task title is empty in the first place. However, in that case, it will not be consistent with the pre-existing modal.

started discussion at https://chat.zulip.org/#narrow/stream/6-frontend/topic/description.20field.20in.20todo.20modal/near/1792835